File: app/src/main/java/dev/voleum/ordermolder/viewmodels/ObjectsCashReceiptListItemViewModel.java
package dev.voleum.ordermolder.viewmodels;

import androidx.databinding.Bindable;

import dev.voleum.ordermolder.models.TableObjects;

public class ObjectsCashReceiptListItemViewModel extends AbstractDocListItemViewModel<TableObjects, CashReceiptViewModel> {

    public ObjectsCashReceiptListItemViewModel(TableObjects table, CashReceiptViewModel viewModel) {
        super(table, viewModel);
    }

    @Bindable
    public String getObjectName() {
        return getTable().getObjName();
    }

    @Bindable
    public void setSum(String sum) {
        getTable().setSum(Double.parseDouble(sum));
        getViewModel().countSum();
    }
}

File: app/src/main/java/dev/voleum/ordermolder/viewmodels/ObjectsCashReceiptListItemViewModel.kt
package dev.voleum.ordermolder.viewmodels

import androidx.databinding.Bindable
import dev.voleum.ordermolder.models.TableObjects

class ObjectsCashReceiptListItemViewModel(table: TableObjects, viewModel: CashReceiptViewModel) : AbstractDocListItemViewModel<TableObjects, CashReceiptViewModel>(table, viewModel) {

    override var sum: String
        get() = super.sum
        @Bindable set(value) {
            table.sum = value.toDouble()
            viewModel.countSum()
        }
}